content/blog/aws-eks-managed-nodes-fargate/index.md

```diff
@@ -0,0 +1,310 @@
+
+To opt-in to using Managed Node Groups, you can use the raw [`aws.eks.NodeGroup`](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/aws/eks/#NodeGroup) class. However, if you're using the `eks.Cluster` abstraction, we've made it easy to use them, with the addition of a new `createManagedNodeGroup` function.
+
+For instance, this code creates a new EKS cluster, disabling the default node group &mdash; using `skipDefaultNodeGroup: true`, since we will create our own &mdash; and creates a single Managed Node Group with an IAM role, the desired scaling parameters, instance types, labels, and more:
+
+```typescript
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+
+// Create a VPC for our cluster.
+const vpc = new awsx.ec2.Vpc("my-vpc");
+
+// IAM roles for the node group.
+const role = new aws.iam.Role("my-cluster-ng-role", {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+        Service: "ec2.amazonaws.com",
+    }),
+});
+let counter = 0;
+for (const policyArn of [
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+]) {
+    new aws.iam.RolePolicyAttachment(`my-cluster-ng-role-policy-${counter++}`,
+        { policyArn, role },
+    );
+}
+
+// Create an EKS cluster.
+const cluster = new eks.Cluster("my-cluster", {
+    skipDefaultNodeGroup: true,
+    vpcId: vpc.vpcId,
+    publicSubnetIds: vpc.publicSubnetIds,
+    privateSubnetIds: vpc.privateSubnetIds,
+    deployDashboard: false,
+    instanceRoles: [ role ],
+});
+
+// Create a simple AWS managed node group using a cluster as input.
+const managedNodeGroup = eks.createManagedNodeGroup("my-cluster-ng", {
+    cluster: cluster,
+    nodeGroupName: "aws-managed-ng1",
+    nodeRoleArn: role.arn,
+    scalingConfig: {
+        desiredSize: 1,
+        minSize: 1,
+        maxSize: 2,
+    },
+    instanceTypes: "t2.medium",
+    labels: { "ondemand": "true" },
+    tags: { "org": "pulumi" },
+}, cluster);
+
+// Export the cluster's kubeconfig.
+export const kubeconfig = cluster.kubeconfig;
+```
+
+This can be provisioned with a single `pulumi up`, just like before, but instead of the automatically created default node group, our cluster will use a Managed Node Group instead.
+
+> This looks more complex than the basic `eks.Cluster` example above simply because that exaple benefitted greatly from the EKS package's internal smarts. This approach gives you more control over the worker nodes, while still leaning on EKS to handle the EC2 node management based on your declarative specifications.
```

Comment:
I'd suggest we use the simplified example here to reduce verbosity:

https://github.com/pulumi/pulumi-eks/blob/e9d4c42036c56c7fd526ab6c7ddd85ff10d0fb17/nodejs/eks/examples/managed-nodegroups/index.ts#L19-L24

Reply:
I considered it ... However, that particular example doesn't show any of the control that might drive you to want to use a managed node group, though, does it? (If there's a benefit to it, it's too subtle for me!)

Reply:
Actually, can you help me with one question? What is the essential difference from an end user's perspective between that example you link to, and this one, for purposes of the blog post summary? https://github.com/pulumi/pulumi-eks/blob/e9d4c42036c56c7fd526ab6c7ddd85ff10d0fb17/nodejs/eks/examples/nodegroup/index.ts#L33-L40

Reply:
I wanted to show unmanaged node groups => complex, managed node groups => so much simpler, but the two look essentially equivalent to my eyes...

Reply:
We bake in sane defaults, but still allow complete control possible.

At the very least i think we should not include the note about this being more complex than usual

Reply:
The intended flow of the post was:

1. Complex (EC2)
2. Easier (Managed Node Groups)
3. Easiest (Fargate)

Do you have a good example of (1) that drives the point home? Every example I can find for going from (1) to (2) actually makes them look more complex.

It could be that we show:

0. eks.Cluster with defaults -- pretty darn easy already
1. Complex (EC2)
2. Easier (Managed Node Groups)
3. Easiest (Fargate)

Reply:
Ok, I've given this a shot. Basically, lead in with the simple `eks.Cluster` example. _Then_ go into a section about manual EC2 node groups, with a more complex example. Then the managed node group example is simplified similar to the example you pointed out. Feels like it flows better -- lmk what you think!

Reply:
@joeduffy The updates flow way better 👍